modules/imap/handler_modules.php

```diff
@@ -840,16 +845,40 @@ public function process() {
                     $existingEmails = array_map(function($c){
                         return $c->value('email_address');
                     },$contact_list);
-                    list($total, $results) = $imap->get_mailbox_page(hex2bin($form['folder']), $sort, $rev, $filter, $offset, $limit, $keyword, $existingEmails, $include_preview);
+                    list($total, $results) = $imap->get_mailbox_page(hex2bin($form['folder']), $sort, $rev, $filter, $offset, $limit, $keyword, $existingEmails, $include_content_body);
                 } else {
-                    list($total, $results) = $imap->get_mailbox_page(hex2bin($form['folder']), $sort, $rev, $filter, $offset, $limit, $keyword, null, $include_preview);
+                    list($total, $results) = $imap->get_mailbox_page(hex2bin($form['folder']), $sort, $rev, $filter, $offset, $limit, $keyword, null, $include_content_body);
                 }
                 foreach ($results as $msg) {
                     $msg['server_id'] = $form['imap_server_id'];
                     $msg['server_name'] = $details['name'];
                     $msg['folder'] = $form['folder'];
+                    $uid = $msg['uid'];
+
+                    if ($ceo_use_detect_ceo_fraud && hex2bin($form['folder']) == 'INBOX') {
+                        if ($this->isCeoFraud($msg['to'], $msg['subject'], $msg['preview_msg'])) {
+                            
+                            $folder = "Suspicious emails";
+                            if (!count($imap->get_mailbox_status($folder))) {
+                                $imap->create_mailbox($folder);
+                            }
+                            $dest_folder = bin2hex($folder);
+                            $server_ids = array(
+                                $form['imap_server_id'] => [
+                                    $form['folder'] => $uid
+                                ]
+                            );
+                            imap_move_same_server($server_ids, "move", $this->cache, [null, null, $dest_folder]);
+                            $msg = [];
+                            $total--;
+                        }
+                    }
+                    if (! $include_preview && isset($msg['preview_msg'])) {
+                        $msg['preview_msg'] = "";
+                    }
                     $msgs[] = $msg;
```

Comment:
Please refrain from adding empty messages to the list. If you skip the message, simply don't add it to the list and update the total to match. Then, we won't need to change calling code to check if the msg is empty or not.

Reply:
Oh yes!

thanks.